src/mongocxx/exception/base.hpp

```diff
@@ -30,17 +27,15 @@ namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace exception {
 
-using error_and_code_type = std::tuple<std::string, std::int32_t>;
-
 class MONGOCXX_API base : virtual public std::exception {
    public:
-    base(bsoncxx::document::value raw_server_error);
-    base(error_and_code_type error_and_code);
+    base(std::error_code ec);
+
+    base(std::error_code ec, std::string what_arg);
```

Comment:
can you just give 'what' a default value here, and for the 'what' overload below two? I think you can get this down to two constructors.